### DISCUSSION TITLE
Same arg in default args for different operators

### DISCUSSION THREAD

**Discussion by JavierLopezT** [apache/airflow#16045] [Ideas]
Sometimes, different operators share the same argument. For instance, SnowflakeOperator and MySqlOperator have both the argument `database`. It would be nice to be able to assign a different value for the arg in `default_args` depending on the operator. So the `default_args` dict would be something like this:

```
default_args = {
'email': blabla,
'database': {'SnowflakeOperator': 'snow_database',
         'MySqlOperator': 'mysql_database'}
}
```

That way, if you have a few of both operators, you don't have to type the database argument for all the operators of one kind. WDYT? Would this feature makes sense?

**Comment by potiuk** [comment]
That's an interesting feature I think. But I think it's a bit unreadable this way as there is slightly illogical sequence of grouping the parameters. It would have been much better if we had a new argument like:

```
default_task_args = {
  'SnowflakeOperator': {
     'database': 'snow_database',
     'another_arg': 'another_value"
  }
  'MySqlOperator': {
    "database": 'mysql_database'
  }
}
```

**Reply by potiuk** [reply]
Also we would need to define behaviour of the edge cases: 
*  inheritance (if SnowflakeOperator derives from SQLOperator - should it get "SQLOperator"'s args? What if the args are defined for both "SQL" and "Snowflake"
* how to define the operator "keys" . I.e. what should happen if we have same operator names in different packages. Should we make the keys "full class including package" ?

**Reply by uranusjr** [reply]
I kind of feel this is complicating the design too much. Would it be better to encourage operator subclassing instead? (Or is subclassing a bad idea to begin with?)

```python
from airflow.operators.mysql import MySqlOperator as _MySqlOperator

class MySqlOperator(_MySqlOperator):
    def __init__(self, **kwargs):
        super().__init__(daabase="mysql_database", **kwargs)

with DAG(...) as dag:
    mysql = MySqlOperator(...)
    ...
```

Maybe we can provide some factory functions to make the subclassing simpler?
